doc/designs/fips_indicator.md

```diff
@@ -26,59 +26,122 @@ The following information was extracted from the FIPS 140-3 IG [5] “2.4.C Appr
 - Testing is required to execute all services and verify that the indicator provides an unambiguous indication of whether the service utilizes an approved cryptographic algorithm, security function or process in an approved manner or not.
 - The Security Policy may require updates related to indicators. AWS/google have added a table in their security policy called ‘Non-Approved Algorithms not allowed in the approved mode of operation’. An example is RSA with a keysize of < 2048 bits (which has been enforced by [4]).
 
+Since any new FIPS restrictions added could possibly break existing applications
+the following additional OpenSSL requirements are also needed:
+
+- The FIPS restrictions should be able to be disabled using Configuration file options (This results in unapproved mode and requires an indicator).
+- A mechanism for logging the details of any unapproved mode operations that have been triggered (e.g. DSA Signing)
+- The FIPS restrictions should be able to be enabled/disabled per algorithm context.
+- If the per algorithm context value is not set, then the  Configuration file option is used.
+
 Solution
 --------
 
-We already have most of the existing code in the FIPS provider using
+In OpenSSL most of the existing code in the FIPS provider is using
 implicit indicators i.e. An error occurs if existing FIPS rules are violated.
 
 The following rules will apply to any code that currently is not FIPS approved,
 but needs to be.
 
-- There will be a 'fips approved' mode per provider operation context that is on by default.
-- In this mode any FIPS rule that is violated will result in an error.
-- The 'fips approved' mode may be turned off per context via a OSSL_PARAM setter, if it
-is turned off then the operation is not FIPS approved.
-- A getter is required that returns if the operation is 'fips approved' using an
-OSSL_PARAM. The returned value can be a combination of the set 'fips_approved' mode
-plus any other logic. This is an explicit indicator.
-- Any algorithm that is transitioning [4] to not being allowed should be removed from
-the fips provider.
-- If an algorithm is transitioning [4] to be only allowed for processing
-(e.g. verification, signing, key validation) then the protection code
-(keygen, signing, encryption) should be removed from the fips provider, any
-attempt to use the protection API's should result in an error. The processing
-code should still be functional.
-
-Other rules:
-The existing flags that we set in the fips config file to control security checks
-will continue to function as they do now, and will not be affected by the
-strict mode variable.
-The getter to determine if we are fips approved will however take the flags
-into account.
-
-Motivation
+- The fipsinstall application will have a configurable item added for each algorithm that requires a change.
```

Comment:
Is this means that my PR like https://github.com/openssl/openssl/pull/23889 should add several options such as `no-sshkdf-check`, `no-hkdf-check` and etc.?

Reply:
Until this PR is approved and merged, I dont recommend doing anything just yet... :)

Reply:
I know that :)

Just curious if there will be a lot of configurable items in the future. But I will wait for this PR approved before doing something.

Reply:
If those algorithms each have special checks that have been added, that are likely to break things, then yes they would need a flag.

Reply:
Got it. Thanks for explanation.